### DISCUSSION TITLE
Are Provider IDs global like MDS, or per city?

### DISCUSSION THREAD

**Discussion by schnuerle** [openmobilityfoundation/curb-data-specification#24] [Data Specification]
How are provider/operator IDs handled? 

1. Globally unique like MDS in a file on the CDS GitHub repo? 
2. Locally unique and managed by each agency as fleet registration happens? 

Does every company get one ID, or could company divisions (Freight, Ground, etc) have their own IDs?

**Comment by schnuerle** [comment]
Per the discussions, it seems the working group is leaning toward:

1. Globally unique identifiers like in MDS providers.csv, to make things easier for commercial companies in multiple cities and simplicity across jurisdictions

**Reply by jfh01** [reply]
I'm concerned about how a database like this gets maintained. Many curb users have multiple companies/contractors delivering on their behalf. Who decides whether FedEx and FedEx Ground are two different IDs? How about the distributor that runs trucks for multiple brands?

**Reply by schnuerle** [reply]
If we keep it global, I think it can handle these scenarios the same way that MDS does does. The companies get to decide how they are identified (eg FedEx could get an id, or FedEx can get one and FedEx Ground can get another) and subsidiaries or different branding can also get their own IDs even if run by a parent company.

We have [documentation on how to do this now for MDS](https://github.com/openmobilityfoundation/mobility-data-specification/wiki/Adding-an-MDS-Provider-ID) which we can adapt for CDS, and it's 1) pretty fast for an operator to make an issue using the template and 2) fast for OMF staff to review for accuracy and look for any issues and then push to the `dev` branch where the ID can begin to be used in production.

Doing it globally also helps know who is doing what in the CDS space, reduces complexity for operators since they can reference just one or more globally static IDs and only have to create them once, and allows agencies to lookup operators by name in a single online location.

EDIT: One difference here is we may end up having more operators in the CDS space than MDS, though it remains to be seen.

**Reply by jlarsonOmahaNE** [reply]
I think that there should be both a global and local database. In Omaha we have more local delivery companies than global but also have local companies that deliver on the behalf of the global companies and some that delivery their own products. Would a global database also have a subset for local providers?

**Reply by schnuerle** [reply]
The spec could be made clear that ids are optional unless required by a city, and if you have more local, smaller operators then you can use the provider_name field to keep track of them in CDS.

**Reply by jlarsonOmahaNE** [reply]
I agree with using the provider_name field for smaller/local operators and would work for our use cases

**Comment by jiffyclub** [comment]
+1 for globally unique IDs. Makes things a lot easier for us at Populus to track providers if they are all listed in one place. It also simplifies cases like where there are overlapping regions of interest, like a city and a surrounding county or MPO. If we need disambiguation I'm all for adding metadata to the providers.csv, like address, phone number, website, description, etc.

**Comment by schnuerle** [comment]
Here's how we are going to handle it, with provider_id . See `provider_id` and `provider_name` definitions [here](https://github.com/openmobilityfoundation/curb-data-specification/blob/feature-release-work-1/events/README.md#curb-event):

Name | Type | Required/Optional | Description
-- | -- | -- | --
provider_id | UUID | Optional | Unique ID of the provider/operator/company responsible for operating the vehicle at the time of the event, if any. IDs are global and come from the providers.csv file here in the CDS repo. An agency a their discretion may allow a small, local company to simply provide a consistent provider_name string instead of a provider_id from the global providers.csv file.
provider_name | String | Optional | Name of the provider responsible for operating the vehicle, device, or sensor at the time of the event. May be sent along with provider_id or on its own.
